docs/how_to/nginx_https.md

```diff
@@ -0,0 +1,49 @@
+# How to use Nginx to serve Livebook over HTTPS
+
+This guide shows you how to use Nginx to serve Livebook over HTTPS.
+
+## Prerequisites
+
+- Nginx installed
+- Livebook installed
+- Both Livebook and Nginx running on the same machine or within the same network
+- SSL certificate and key files
+
+## Nginx configuration
+
+Use the following Nginx config file as a starting point:
+
+```nginx
+http {
+    server {
+        listen 443 ssl;
+        server_name your_domain_or_ip;  # e.g., livebook.example.com
+
+        ssl_certificate /path/to/your/ssl_certificate.crt;  # e.g., /etc/nginx/ssl/livebook.crt
+        ssl_certificate_key /path/to/your/ssl_certificate.key;  # e.g., /etc/nginx/ssl/livebook.key
+
+        location / {
+            proxy_pass http://livebook_ip:livebook_port;  # e.g., http://172.20.0.3:8080 (Livebook's default port is 8080)
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade $http_upgrade;
+            proxy_set_header Connection "upgrade";
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+        }
+    }
+
+    server {
+        listen 80;
+        server_name your_domain_or_ip;  # e.g., livebook.example.com
+        return 301 https://$host$request_uri;
+    }
+}
+
+events {}
+```
+
+Note that there's no need to configure a proxy for Livebook's iframe port (the default is 8081)
+because when accessing Livebook over HTTPS, it loads its iframe from
+https://livebookusercontent.com, not from your installed Livebook instance.
```

Comment:
Do we need to say this? I am generally not a big fan of telling people what they should _not_ do.

Reply:
I thought about mentioning that because the user that asked me for help while trying to figure this out by himself was confused if he needed to map the iframe port as well.

Here's a quote from him:

> ---- Now
> Client {8080, 8081} -> Livebook Docker {8080, 8081}
>
> ---- What I need to achieve
> Client {443} -> nginx {443} -> Livebook Docker {8080, 8081}
>
> ---- What I haven't fathomed out yet is should it be?
Client {8443, 8444} -> nginx {8443, 8444} -> Livebook Docker {8080, 8081}
> 
> If the above can be achieved I need to check how the browser handles the certificate as not on usual port 443 It should be OK but you never know…

But not that we have a how-to guide, maybe there is no need to explain that there's no need to map the iframe port.

What do you think? I'll go with your suggestion on this one, no strong opinion. Probably no need to mention indeed.

Reply:
Perhaps it could be better phrased in the positive sense as there is some merit in mentioning the absence of port `8081`. How about;

> Since this is a configuration for HTTPS, Livebook takes care of the `iframe` port (8081) for us - see [the documentation](https://github.com/livebook-dev/livebook/blob/v0.13.3/README.md?plain=1#L252-L254).